packages/Tag/src/TeamColorIcon.tsx

```diff
@@ -0,0 +1,148 @@
+import * as React from 'react';
+
+import cx from 'classnames';
+
+import * as variables from '@igloo-ui/tokens/dist/base10/tokens.json';
+
+import './team-color-icon.scss';
+
+export type Size = 'small' | 'medium' | 'large' | 'xlarge';
+
+export interface TeamColorIconProps extends React.ComponentProps<'div'> {
+  /** The content of the team color icon */
+  children?: React.ReactNode;
+  /** Add a specific class to the team color icon */
+  className?: string;
+  /** The background color of the team icon */
+  teamColor?: string;
+  /** Specifies the size of the team icon */
+  size?: Size;
+  /** Specifies the id of the color tag,
+   * if the team color properties is empty,
+   * we will calculate the color with the id. */
+  teamId?: string;
+  /** Sepcifies the name of the team color,
+   * we will create the letter composition with it */
+  teamName?: string;
+}
+
+const TeamColorIcon: React.FunctionComponent<TeamColorIconProps> = ({
+  children,
+  className,
+  teamColor,
+  size = 'small',
+  teamId,
+  teamName,
+}) => {
+  const classes = cx(
+    'ids-team-color-icon',
+    `ids-team-color-icon--${size}`,
+    className
+  );
+  const allOrganisationId = '00000000-0000-0000-0000-000000000000';
```

Comment:
The whole logic part that defines a color according to the id should be the responsibility of the user ex. portal. By doing so we make the component too specific to a use case. The idea here is to receive a color ex. #ccff22 and pass it to the square.

Reply:
alright, so I'll just remove the ID prop and its logic?

Reply:
yes, the ID and logic